src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +291,220 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
+        private (Command, IEnumerable<ValidateSymbol<CommandResult>>) CreateSubmitEntryPointCommand(IEntryPoint entryPoint)
+        {
+            var validators = Enumerable.Empty<ValidateSymbol<CommandResult>>();
+
+            var command = new Command(entryPoint.Name, entryPoint.Summary)
+            {
+                Handler = CommandHandler.Create((ParseResult parseResult, AzureSettings settings) => this.Submit(parseResult, settings, entryPoint))
+            };
+            foreach (var option in entryPoint.Options)
+            {
+                command.AddOption(option);
+            }
+
+            validators = validators.Concat(AddOptionIfAvailable(command, SubscriptionOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, ResourceGroupOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, WorkspaceOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, this.CreateTargetOption(entryPoint)));
+            validators = validators.Concat(AddOptionIfAvailable(command, StorageOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, AadTokenOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, BaseUriOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, LocationOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, JobNameOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, ShotsOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, OutputOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, DryRunOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, VerboseOption));
+            validators = validators.Concat(MarkOptionsAsMutuallyExclusive(
+                command,
+                new[] { BaseUriOption.Aliases.First(), LocationOption.Aliases.First() }));
+
+            return (command, validators);
+        }
 
         /// <summary>
-        /// The output option.
+        /// Creates the simulation option for the simulate command with the given entry point.
         /// </summary>
-        internal static readonly OptionInfo<OutputFormat> OutputOption = new OptionInfo<OutputFormat>(
-            ImmutableList.Create("--output"),
-            OutputFormat.FriendlyUri,
-            "The information to show in the output after the job is submitted.");
+        /// <param name="entryPoint">The entry point.</param>
+        /// <returns>The OptionInfor for the simulator option.</returns>
+        private OptionInfo<string> CreateSimulatorOption(IEntryPoint entryPoint) =>
+            new OptionInfo<string>(
+                this.settings.SimulatorOptionAliases,
+                entryPoint.DefaultSimulatorName,
```

Comment:
The default simulator option applies to the entire project, rather than for a specific entry point. Maybe it could be moved out of `IEntryPoint` into `DriverSettings`?

Reply:
Do we not support passing the desired simulator into the Entry Point attribute in Q# anymore? I thought that was what set the default simulator per entry point.

Reply:
No, the entry point attribute has `Unit` as its underlying type - see #451.

Reply:
Oh okay, I believe I was mistakenly thinking of the Unit Test attribute.

Reply:
I was able to do this, along with moving the CreateDefaultCustomSimulator function to the driver settings. The two had to be moved together. I think this change helps keep the entry point classes tightly defined. Thanks!